SmartSaver.Domain/Services/DocumentServices/PDFCreator.cs

```diff
@@ -52,91 +48,76 @@ public byte[] GeneratePDF(IEnumerable<Transaction> transaction, DateTime from, D
-            int serialNumber = 1;
-            for (int i = 0; i < 5; i++) // changed with: foreach(var item in totalIncomeByCategory)
+            #region Block creation methods
+            TextBlock tableHead = delegate(string text, float fontSize) // Formatting based on TABLE HEADER
             {
-                _pdfPCell = new PdfPCell(new Phrase(serialNumber++.ToString(), _fontStyle));
+                _fontStyle = FontFactory.GetFont("Tahoma", fontSize, 1);
+                _pdfPCell = new PdfPCell(new Phrase(text));
                 _pdfPCell.HorizontalAlignment = Element.ALIGN_CENTER;
                 _pdfPCell.VerticalAlignment = Element.ALIGN_MIDDLE;
-                _pdfPCell.BackgroundColor = BaseColor.WHITE;
+                _pdfPCell.BackgroundColor = BaseColor.LIGHT_GRAY;
+                _pdfPCell.ExtraParagraphSpace = 0;
                 _pdfTable.AddCell(_pdfPCell);
+            };
 
-                _pdfPCell = new PdfPCell(new Phrase(i.ToString(), _fontStyle)); // value changed with: item.Key.ToString()
+            TextBlock tableBody = delegate (string text, float fontSize) // Formatting based on TABLE BODY
+            {
+                _fontStyle = FontFactory.GetFont("Tahoma", fontSize, 0);
+                _pdfPCell = new PdfPCell(new Phrase(text, _fontStyle));
                 _pdfPCell.HorizontalAlignment = Element.ALIGN_CENTER;
                 _pdfPCell.VerticalAlignment = Element.ALIGN_MIDDLE;
                 _pdfPCell.BackgroundColor = BaseColor.WHITE;
                 _pdfTable.AddCell(_pdfPCell);
+            };
+            #endregion
 
-                _pdfPCell = new PdfPCell(new Phrase(i.ToString(), _fontStyle));  // value changed with: item.Value.ToString()
-                _pdfPCell.HorizontalAlignment = Element.ALIGN_CENTER;
-                _pdfPCell.VerticalAlignment = Element.ALIGN_MIDDLE;
-                _pdfPCell.BackgroundColor = BaseColor.WHITE;
-                _pdfTable.AddCell(_pdfPCell);
+            #region Table header
+            tableHead("ID", 8f);
+            tableHead("Category", 8f);
+            tableHead("Amount", 8f);
+            _pdfTable.CompleteRow();
+            #endregion
 
+            #region Table body
+            int serialNumber = 1;
+            foreach(var expense in totalExpenseByCategory) // changed with: foreach(var item in totalIncomeByCategory)
+            {
+                tableBody(serialNumber.ToString(), 8f);
+                tableBody(expense.Key.ToString(), 8f);
+                tableBody(expense.Value.ToString(), 8f);
+                _pdfTable.CompleteRow();
+                serialNumber++;
+            }
+            if(totalExpenseByCategory.Count == 0)
+            {
+                tableBody("no data to display", 8f);
+                tableBody("no data to display", 8f);
+                tableBody("no data to display", 8f);
                 _pdfTable.CompleteRow();
             }
```

Comment:
What's the purpose to fill table with this kind of data? Just don't create pdf if there are no such transactions and provide a useful message for user. Or just create an empty pdf